### PR TITLE
CI/Build: Add Gradle Build Scans

### PR DESCRIPTION
**Pull request by snazy**
Gradle Build Scans, free of use, collect a lot of information about a Gradle build, including the actual output of failed test. This becomes quite convenient when inspecting test failures in CI and a lot of other information about Gradle builds.

[Example build scan](https://scans.gradle.com/s/jpuykotf4hac6)

### CODE REVIEW

**Comment by jbonofre** [comment]
@snazy what about using ASF Develocity ? It's hosted here: https://ge.apache.org/ and "hosted"/approve by the ASF.

I would prefer to use ASF hosted platform here.

**Comment by jbonofre** [comment]
For context: https://infra.apache.org/gradle.html

**Comment by snazy** [comment]
Good point, updated the PR.

**Comment by adutra** [comment]
> Well, Apache's instance rejects unauthenticated build scans, which is always the case for PR CI runs (no secrets in those GH workflows). So I think it's fine to let CI runs that don't have the access-key secret publish to Gradle's infra, and those that have it publish to Apache's infra.

Why can't we configure our PR CI runs to authenticate? The section named "GitHub Actions" in [in this page](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=263426153#ProjectOnboardingInstructionsforDevelocity-GitHubActions) says:

> For GitHub Actions builds in the Apache GitHub organization, an access token is stored in the organizational secret `GE_ACCESS_TOKEN`.

**Comment by jbonofre** [comment]
Two comments:
- we need approval from PPMC and security team to export data in a third party instance
- as other projects (like Apache Beam or Pekko) are using ge.apache.org, I wonder why we would be different 😄

**Comment by jbonofre** [comment]
See https://cwiki.apache.org/confluence/pages/viewpage.action?spaceKey=INFRA&title=Project+Onboarding+Instructions+for+Develocity

**Comment by snazy** [comment]
>     * as other projects (like Apache Beam or Pekko) are using ge.apache.org, I wonder why we would be different 😄

That still requires all GH workflow runs to have access to secrets, which is a big concern IMHO.

**Comment by jbonofre** [comment]
@snazy For GitHub Actions builds in the Apache GitHub organization, an access token is stored in the organizational secret `GE_ACCESS_TOKEN`. For security, GitHub Actions do not pass secrets to runners running workflows triggered from forked repositories.
I don't see the issue here as the `GE_ACCESS_TOKEN` is setup by INFRA.
For local build, users can use their Apache ID.

**Comment by snazy** [comment]
@jbonofre GH actions run from forks do _NOT_ have any secrets, that's how GH workflows work.

**Comment by jbonofre** [comment]
@snazy yes, that's expected. That's why I don't the security question :) If an user forks Polaris, if he's a Apache committer, he can use his Apache account, else he doesn't push.

Sorry, I failed to see the point 😄

**Comment by snazy** [comment]
Yes, you failed the point. See the _warning_ box under the discussion of [`pull_request_target`](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target) vs the [`pull_request`](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request) trigger. Apache Committer or not does _not_ matter here.

**Comment by snazy** [comment]
TL;DR both scenarios, `pull_request_target` for CI _against_ `apache/polaris` and using branches _in_ the `apache/polaris` repo with `pull_request` are _both_ huge security risks - see [this post](https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/).

**Comment by jbonofre** [comment]
My point about Apache Committer is for local build (not related to GH Action).

I think I understand your point about PR, but I was talking about "regular" GH Action build.
For PRs, are you proposing to systematically publish build scans ?

**Comment by snazy** [comment]
> For PRs, are you proposing to systematically publish build scans ?

Yes, that's basically the whole point - I want to inspect build scans for PRs, not just test results, but more than that.

**Comment by jbonofre** [comment]
@snazy so like this one: https://ge.apache.org/s/7hwlge7hwhch4
It's a build on a PR (https://github.com/apache/beam/pull/32400) on Beam (action https://github.com/apache/beam/actions/runs/12012626739).

So it seems to work on Apache Beam.

**Comment by jbonofre** [comment]
@snazy I'm checking this.

**Comment by snazy** [comment]
Any updates on this @jbonofre ?

**Comment by snazy** [comment]
Build scans are _very_ useful and _much_ more convenient than downloading some archive, unpacking it, finding the right files, etc.
It's worth tackling this effort.

**Comment by snazy** [comment]
Closed in favor of #2559
